lib/python/machinekit/service.py

```diff
@@ -105,16 +126,32 @@ def __init__(
         self.type = type_
         self.port = port
         self.name = name
+        self.headline = headline
         self.host = host
         self.loopback = loopback
         self.debug = debug
 
         self.stype = '_machinekit._tcp'
         self.subtype = '_%s._sub.%s' % (self.type, self.stype)
 
+        if announce_format is None:
+            announce_format = 'MK $SRVNAME on $HOSTNAME'
+
         if name is None:
-            pid = os.getpid()
-            self.name = '%s service on %s pid %i' % (self.type.title(), self.host, pid)
+            self.name = self.type.title()
+
+        formatdict = {
+            "$MKUUID" : self.svc_uuid,
+            "$HOSTNAME" : self.host,
+            "$SRVNAME" : self.name,
+            "$SRVTYPE" : self.type
+        }
+
+        if headline is None:
+            self.headline = self.__multi_replace(announce_format, formatdict, True)
+
+        # Truncate the headline to fit in the DNS length limit
+        self.headline = self.headline[:63]
```

Comment:
I figured truncation was required so we don’t push an invalid domain name to avahi. Failing with an error is fine also, I can update the United comments with instructions on length to help when an error occurs.

Reply:
Yes, it is required. At least by the standards I have read. I was just expecting to fail. I won't push my worldview on this, but it would require to expect for example just part of GUID on the part of a remote user (or programmer or remote component).

Reply:
Added commit to change this behavior to error on invalid.

Reply:
Thank you.